woocommerce/woocommerce-ios
# [Tablet Orders] Persist in-flight order details on size class changes

Closes: #12252

## Description
This PR addresses the issue of modal views being dismissed automatically on size class changes, which produces a loss of the data the merchant may already have entered, but not specifically saved.

We tackle the cases for `Customer Notes` and `Customer Details`, these are the ones where a customer might have written more data manually into the fields and prone to be lost upon accidental dismissal.

**Question:** I've been experimenting with applying the same behaviour for Custom Amounts, however, I'm not convinced is the right way to go and perhaps we should just allow the custom amount to be lost upon dismissal? If we do, I believe we can tackle it separately, since we may need to rethink the current implementation of custom amounts as it relies on being re-created and reset multiple times. The work-in-progress experimental branch can be found here: `experiment/12252-continuation-persist-custom-amount-on-size-class-change`

## Changes
- When we initialize an `EditableOrderViewModel`, we'll observe and keep a reference to the latest changes to the customer details object. If there's a size class change triggered from the `OrderForm` then we'll save this latest value to the order.

- With order notes, we just perform a note update when the size class change is triggered.

## Testing instructions
1. Launch the app on an iPad
2. Open the Orders tab and tap `+`
3. Tap `Add Customer Details`
4. Enter some text
5. Change the size class using a split view so that only one pane is shown
6. Observe that the modal view has been dismissed automatically, but the data has persisted in the order
7. Repeat the process for `Add Note`

## Screenshots

![Simulator Screen Recording - iPad Pro (12 9-inch) (6th generation) - 2024-03-20 at 10 02 35](https://github.com/woocommerce/woocommerce-ios/assets/3812076/5e5663c1-6ff9-41ec-a349-4077d0915a60)
